compendium/DeclarativeServices/src/manager/ConfigurationNotifier.cpp

```diff
@@ -0,0 +1,216 @@
+  std::shared_ptr<ComponentConfigurationImpl> mgr)
+{
+  cppmicroservices::ListenerTokenId retToken = ++tokenCounter;
+
+  {
+    auto listener = Listener(notify, mgr);
+
+    auto listenersMapHandle = listenersMap.lock();
+    auto iter = listenersMapHandle->find(pid);
+    if (iter != listenersMapHandle->end()) {
+      (*(iter->second)).emplace(retToken, listener);
+    } else {
+      auto tokenMapPtr = std::make_shared<TokenMap>();
+      tokenMapPtr->emplace(retToken, listener);
+      listenersMapHandle->emplace(pid, tokenMapPtr);
+    }
+  }
+
+  return retToken;
+}
+
+void ConfigurationNotifier::UnregisterListener(
+  std::string pid,
+  const cppmicroservices::ListenerTokenId token) noexcept
+{
+  auto listenersMapHandle = listenersMap.lock();
+  if (listenersMapHandle->empty() || pid.empty()) {
+    return;
+  }
+
+  auto iter = listenersMapHandle->find(pid);
+  if (iter == listenersMapHandle->end()) {
+    return;
+  }
+
+  auto tokenMapPtr = iter->second;
+  for (const auto& tokenMap : (*tokenMapPtr)) {
+    if (tokenMap.first == token) {
+      tokenMapPtr->erase(tokenMap.first);
+      if (tokenMapPtr->size() == 0) {
+        listenersMapHandle->erase(iter);
+      }
+      break;
+    }
+  }
+}
+bool ConfigurationNotifier::AnyListenersForPid(std::string pid) noexcept
+{
+  std::string factoryName;
+  std::shared_ptr<ComponentConfigurationImpl> mgr;
+  {
+    auto listenersMapHandle = listenersMap.lock();
+    if (listenersMapHandle->empty() || pid.empty()) {
+      return false;
+    }
+
+    auto iter = listenersMapHandle->find(pid);
+    if (iter != listenersMapHandle->end()) {
+      return true;
+    }
+
+    //The exact pid isn't present. See if this is a factory pid.
+    auto position = pid.find('~');
+    if (position == std::string::npos) {
+      return false;
+    }
+    //This is a factoryPid with format factoryComponentName~instanceName.
+    //See if we're listening for changes to this factoryComponentName
+    factoryName = pid.substr(0, position);
+    if (factoryName.empty()) {
+      return false;
+    }
+    iter = listenersMapHandle->find(factoryName);
+    if (iter == listenersMapHandle->end()) {
+      return false;
+    }
+    auto listener = iter->second->begin();
+
+    mgr = listener->second.mgr;
+    if (mgr->GetMetadata()->factoryComponentID.empty()) {
+      // The component in our listener's map is not a factory component.
+      return false;
+    }
+  } //release listenersMapHandle lock
+  CreateFactoryComponent(factoryName, pid, mgr);
+  return true;
+}
+void ConfigurationNotifier::CreateFactoryComponent(
+  const std::string factoryName,
+  std::string pid,
+  const std::shared_ptr<ComponentConfigurationImpl> mgr)
+{
+  auto oldMetadata = mgr->GetMetadata();
+  auto newMetadata = std::make_shared<ComponentMetadata>(*oldMetadata);
+
+  newMetadata->name = pid;
+  newMetadata->factoryComponentID =
+    ""; // this is a factory instance not a factory component
```

Comment:
Place '= "";' on the same line as "newMetadata->factoryComponentID. Move comment above that line